### PR TITLE
document all the route annotations and PR11984

### PR DESCRIPTION
**Pull request by JacobTanenbaum**

### CODE REVIEW

File: architecture/core_concepts/routes.adoc [hunk 1]
```diff
@@ -817,11 +818,32 @@ resolution order (oldest route wins).
...
 
 
-[[route-specific-timeouts]]
-== Route-specific Timeouts
+[[route-specific-annotations]]
+== Route-specific Annotations
 
-By default, a server-side timeout is set across all routes. It is possible to
-change that timeout using an annotation in the route configuration.
+By default, or using an environment variable a number of configuration options
```
Comment (ramr) [inline architecture/core_concepts/routes.adoc]: Maybe reword this - sounds a bit odd `by default or using` - something like `Using environment variables as defined in <add-link-to-section>, a router can set the default options for all the routes it exposes. An individual route can choose to override some of these defaults by providing specific configuration via/in (I like via but its normally not used in docs) its annotations`.

**Review by ramr — CHANGES_REQUESTED** [review]

**Comment by JacobTanenbaum** [comment]
@ramr does this clear it up?

**Comment by ramr** [comment]
LGTM

**Review by ramr — COMMENTED** [review]
LGTM

**Review by ramr — APPROVED** [review]

File: architecture/core_concepts/routes.adoc [hunk 2]
```diff
@@ -219,6 +220,7 @@ variables on the *deployment config* for the router to alter its configuration.
 |`*ROUTER_SUBDOMAIN*`|  | The template that should be used to generate the hostname for a route without spec.host (e.g. `${name}-${namespace}.myapps.mycompany.com`).
 |`*ROUTER_SYSLOG_ADDRESS*` |  | Address to send log messages. Disabled if empty.
 |`*ROUTER_TCP_BALANCE_SCHEME*` | source | Load-balancing strategy for multiple endpoints for pass-through routes. Available options are `source`, `roundrobin`, or `leastconn`.
+|`*ROUTER_LOAD_BALANCE_ALGORITHM*` | leastconn | Load-balanceing strategy for multiple endpoints for edge and reencrypt terminated routes. Available options are `source`, `roundrobin`, or `leastconn`.
```
Comment (knobunc) [inline architecture/core_concepts/routes.adoc]: Load-balanceing => Load-balancing

Doesn't this also apply to http routes?

File: architecture/core_concepts/routes.adoc [hunk 3]
```diff
@@ -817,11 +819,32 @@ resolution order (oldest route wins).
...
+
+*Route Annotations*
+
+For all the items outlined in this section, you can set annotations on the
+*route definition* for the route to alter its configuration
+
+.Route Annotations
+[cols="2*", options="header"]
+|===
+|Variable | Description
```
Comment (knobunc) [inline architecture/core_concepts/routes.adoc]: Can we sort this section alphabetically by annotation?

File: architecture/core_concepts/routes.adoc [hunk 4]
```diff
@@ -817,11 +819,32 @@ resolution order (oldest route wins).
...
+|`*haproxy.router.openshift.io/balance*`| sets the load balance algorithm. Available options are `source`, `roundrobin`, or `leastconn`.
```
Comment (knobunc) [inline architecture/core_concepts/routes.adoc]: The first letter of the description should be uppercase to match the other table.

Should we indicate the ENV that provides the default (perhaps in a new column)

load-balancing (or make the pervious one match this case)

File: architecture/core_concepts/routes.adoc [hunk 5]
```diff
@@ -817,11 +819,32 @@ resolution order (oldest route wins).
...
+|`*haproxy.router.openshift.io/timeout*` | sets a server side timeout. overwrites `*ROUTER_DEFAULT_SERVER_TIMEOUT*` for that route
```
Comment (knobunc) [inline architecture/core_concepts/routes.adoc]: server-side

Just make the default env be noted in a new column

File: architecture/core_concepts/routes.adoc [hunk 6]
```diff
@@ -817,11 +819,32 @@ resolution order (oldest route wins).
...
+|`*haproxy.router.openshift.io/rate-limit-connections*`| setting to true enables rate limiting functionality
```
Comment (knobunc) [inline architecture/core_concepts/routes.adoc]: Should true be in quotes?  Or in a fixed width font?  What's the style elsewhere?

File: architecture/core_concepts/routes.adoc [hunk 7]
```diff
@@ -817,11 +819,32 @@ resolution order (oldest route wins).
...
+|`*haproxy.router.openshift.io/rate-limit-connections.concurrent-tcp*`| limits the number of concurrent tcp connections shared by an IP address
```
Comment (knobunc) [inline architecture/core_concepts/routes.adoc]: TCP?  What's the style?

File: architecture/core_concepts/routes.adoc [hunk 8]
```diff
@@ -817,11 +819,32 @@ resolution order (oldest route wins).
...
+|`*haproxy.router.openshift.io/rate-limit-connections.rate-tcp*`| limits the rate at which an IP address can make tcp connections
+|`*haproxy.router.openshift.io/rate-limit-connections.rate-http*`| limits the rate at which an IP address can make http requests
```
Comment (knobunc) [inline architecture/core_concepts/routes.adoc]: HTTP?  What's the style?

File: architecture/core_concepts/routes.adoc [hunk 9]
```diff
@@ -817,11 +819,32 @@ resolution order (oldest route wins).
...
+|`*haproxy.router.openshift.io/disable_cookies*`| disables the use of cookies, the balance algorithm is used to determine which backend serves connections everytime
```
Comment (knobunc) [inline architecture/core_concepts/routes.adoc]: Disables the use of cookies to track related connections.  If set to 'true' the balance algorithm is used to choose which back-end serves connections for each HTTP request.

File: architecture/core_concepts/routes.adoc [hunk 10]
```diff
@@ -817,11 +819,32 @@ resolution order (oldest route wins).
...
+|`*router.openshift.io/haproxy.health.check.interval*`| set the health check interval for backend endpoints
```
Comment (knobunc) [inline architecture/core_concepts/routes.adoc]: Sets the interval for the back-end health checks.

**Review by knobunc — CHANGES_REQUESTED** [review]

**Review by knobunc — APPROVED** [review]
LGTM

**Comment by knobunc** [comment]
@JacobTanenbaum What releases should this be for?  Please put that in the PR description.

@openshift/team-documentation PTAL.  This has passed technical review.

**Comment by ahardin-rh** [comment]
LGTM. I added a to_followup label, as I will make some minor follow-up edits to the topic as a whole in a follow-up PR. @JacobTanenbaum Is this for 3.4? Thanks!

**Comment by ahardin-rh** [comment]
@JacobTanenbaum Also, can you please give this a squash?

**Comment by JacobTanenbaum** [comment]
@ahardin-rh  squashed

**Comment by ahardin-rh** [comment]
@JacobTanenbaum Thanks! Can you please check my labels? I am assuming this is for 3.4. is that correct?

**Comment by JacobTanenbaum** [comment]
@knobunc is this for 3.4 or 3.5?

**Comment by knobunc** [comment]
@JacobTanenbaum if all of the annotations are in 3.4, then it can be for 3.4.  Can you confirm if they are?

**Comment by JacobTanenbaum** [comment]
they are in 3.4

**Comment by ahardin-rh** [comment]
[rev_history]
|xref:../architecture/core_concepts/routes.adoc#architecture-core-concepts-routes[Core Concepts -> Routes]
|Added descriptions of all xref:../architecture/core_concepts/routes.adoc#route-specific-annotations[route annotations].
%
